phpBB/phpbb/members/controller/team.php

```diff
@@ -0,0 +1,384 @@
+				),
+			),
+		);
+
+		$result = $this->db->sql_query($this->db->sql_build_query('SELECT', $sql_ary));
+
+		$group_ids = $groups_ary = array();
+		while ($row = $this->db->sql_fetchrow($result))
+		{
+			if ($row['group_type'] == GROUP_HIDDEN && !$this->auth->acl_gets('a_group', 'a_groupadd', 'a_groupdel') && $row['ug_user_id'] != $this->user->data['user_id'])
+			{
+				$row['group_name'] = $this->language->lang('GROUP_UNDISCLOSED');
+				$row['u_group'] = '';
+			}
+			else
+			{
+				$row['group_name'] = $this->group_helper->get_name($row['group_name']);
+				$row['u_group'] = append_sid("{$phpbb_root_path}memberlist.$phpEx", 'mode=group&amp;g=' . $row['group_id']);
+			}
+
+			if ($row['teampage_id'])
+			{
+				// Only put groups into the array we want to display.
+				// We are fetching all groups, to ensure we got all data for default groups.
+				$group_ids[] = (int) $row['group_id'];
+			}
+			$groups_ary[(int) $row['group_id']] = $row;
+		}
+		$this->db->sql_freeresult($result);
+
+		$sql_ary = array(
+			'SELECT'	=> 'u.user_id, u.group_id as default_group, u.username, u.username_clean, u.user_colour, u.user_type, u.user_rank, u.user_posts, u.user_allow_pm, g.group_id',
+
+			'FROM'		=> array(
+				USER_GROUP_TABLE => 'ug',
+			),
+
+			'LEFT_JOIN'	=> array(
+				array(
+					'FROM'	=> array(USERS_TABLE => 'u'),
+					'ON'	=> 'ug.user_id = u.user_id',
+				),
+				array(
+					'FROM'	=> array(GROUPS_TABLE => 'g'),
+					'ON'	=> 'ug.group_id = g.group_id',
+				),
+			),
+
+			'WHERE'		=> $this->db->sql_in_set('g.group_id', $group_ids, false, true),
+
+			'ORDER_BY'	=> 'u.username_clean ASC',
+		);
+
+		/**
+		 * Modify the query used to get the users for the team page
+		 *
+		 * @event core.memberlist_team_modify_query
+		 * @var array	sql_ary			Array containing the query
+		 * @var array	group_ids		Array of group ids
+		 * @var array	teampage_data	The teampage data
+		 * @since 3.1.3-RC1
+		 */
+		$vars = array(
+			'sql_ary',
+			'group_ids',
+			'teampage_data',
+		);
+		extract($this->dispatcher->trigger_event('core.memberlist_team_modify_query', compact($vars)));
+
+		$result = $this->db->sql_query($this->db->sql_build_query('SELECT', $sql_ary));
+
+		$user_ary = $user_ids = $group_users = array();
+		while ($row = $this->db->sql_fetchrow($result))
+		{
+			$row['forums'] = '';
+			$row['forums_ary'] = array();
+			$user_ary[(int) $row['user_id']] = $row;
+			$user_ids[] = (int) $row['user_id'];
+			$group_users[(int) $row['group_id']][] = (int) $row['user_id'];
+		}
+		$this->db->sql_freeresult($result);
+
+		$user_ids = array_unique($user_ids);
+
+		if (!empty($user_ids) && $this->config['teampage_forums'])
+		{
+			$this->template->assign_var('S_DISPLAY_MODERATOR_FORUMS', true);
+			// Get all moderators
+			$perm_ary = $this->auth->acl_get_list($user_ids, array('m_'), false);
+
+			foreach ($perm_ary as $forum_id => $forum_ary)
+			{
+				foreach ($forum_ary as $this->auth_option => $id_ary)
```

Comment:
`$this->auth_option` is not defined and the key is actually unused here so maybe just drop it